src/nvim/statusline.c

```diff
@@ -90,85 +90,6 @@ void win_redr_status(win_T *wp)
   } else if (*p_stl != NUL || *wp->w_p_stl != NUL) {
     // redraw custom status line
     redraw_custom_statusline(wp);
-  } else {
-    schar_T fillchar = fillchar_status(&attr, wp);
-    const int stl_width = is_stl_global ? Columns : wp->w_width;
-
-    get_trans_bufname(wp->w_buffer);
-    char *p = NameBuff;
-    int plen = (int)strlen(p);
-
-    if ((bt_help(wp->w_buffer)
-         || wp->w_p_pvw
-         || bufIsChanged(wp->w_buffer)
-         || wp->w_buffer->b_p_ro)
-        && plen < MAXPATHL - 1) {
-      *(p + plen++) = ' ';        // replace NUL with space
-      *(p + plen) = NUL;          // NUL terminate the string
-    }
-    if (bt_help(wp->w_buffer)) {
-      plen += snprintf(p + plen, MAXPATHL - (size_t)plen, "%s", _("[Help]"));
-    }
-    if (wp->w_p_pvw) {
-      plen += snprintf(p + plen, MAXPATHL - (size_t)plen, "%s", _("[Preview]"));
-    }
-    if (bufIsChanged(wp->w_buffer)) {
-      plen += snprintf(p + plen, MAXPATHL - (size_t)plen, "%s", "[+]");
-    }
-    if (wp->w_buffer->b_p_ro) {
-      plen += snprintf(p + plen, MAXPATHL - (size_t)plen, "%s", _("[RO]"));
-    }
-    (void)plen;
-
-    int n = (stl_width + 1) / 2;
-    int this_ru_col = ru_col - (Columns - stl_width);
-    this_ru_col = MAX(this_ru_col, n);
-    if (this_ru_col <= 1) {
-      p = "<";                // No room for file name!
-      plen = 1;
-    } else {
-      int i;
-
-      // Count total number of display cells.
-      plen = (int)mb_string2cells(p);
-
-      // Find first character that will fit.
-      // Going from start to end is much faster for DBCS.
-      for (i = 0; p[i] != NUL && plen >= this_ru_col - 1;
-           i += utfc_ptr2len(p + i)) {
-        plen -= utf_ptr2cells(p + i);
-      }
-      if (i > 0) {
-        p = p + i - 1;
-        *p = '<';
-        plen++;
-      }
-    }
-
-    grid_line_start(&default_grid, is_stl_global ? (Rows - (int)p_ch - 1) : W_ENDROW(wp));
-    const int off = is_stl_global ? 0 : wp->w_wincol;
-
-    int width = grid_line_puts(off, p, -1, attr);
-    grid_line_fill(off + width, off + this_ru_col, fillchar, attr);
-
-    int NameBufflen = get_keymap_str(wp, "<%s>", NameBuff, MAXPATHL);
-    if (NameBufflen > 0 && this_ru_col - plen > NameBufflen + 1) {
-      grid_line_puts(off + this_ru_col - NameBufflen - 1, NameBuff, -1, attr);
-    }
-
-    win_redr_ruler(wp);
-
-    // Draw the 'showcmd' information if 'showcmdloc' == "statusline".
-    if (p_sc && *p_sloc == 's') {
-      n = this_ru_col - plen - 2;  // perform the calculation here so we only do it once
-      const int sc_width = MIN(10, n);
```

Comment:
This is not something we can replicate through statusline expression right now. unless some sort of flex like system is implemented in stl-expr .